tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -1881,27 +1930,20 @@ static bool CombineBasicTypes(
     *pOutKind = AR_BASIC_MIN10FLOAT;
     break;
   case BPROP_BITS16:
-    if ((uEitherFlags & BPROP_MIN_PRECISION) != 0)
-    {
-      *pOutKind = AR_BASIC_MIN16FLOAT;
-    }
-    else
-    {
+    if (IS_BPROP_16BIT_FLOAT_NO_MIN_PRECISION(uLeftProps) ||
+        IS_BPROP_16BIT_FLOAT_NO_MIN_PRECISION(uRightProps)) {
```

Comment:
This is a case where it might not actually be what's intended.
What should happen: `min16float + min16int` -> `min16float`
What this code does: `min16float + min16int` -> `float16_t` (even if -enable-16bit-types is not enabled!)
I think this should just be `BPROP_MIN_PRECISION & uBothFlags` with then/else reversed.